forge-game/src/main/java/forge/game/combat/Combat.java

```diff
@@ -154,13 +154,20 @@ public void endCombat() {
         lkiCache.clear();
         combatantsThatDealtFirstStrikeDamage.clear();
 
-        //update view for all attackers and blockers
+        //clear tracking for cards that care about "this combat"
         for (Card c : attackers) {
```

Comment:
- thanks, but can you explain what you mean by "could be used offensively"?
- Seems to me like `PhaseHandler.endCombat()` calls `Combat.endCombat()` anyway, so I don't understand yet how moving these parts around would fix something?
- also, before it was iterating over all battlefield cards. I think there's a risk you would miss updating some now if they get removed from combat before

Reply:
The actual bug was that clearAttackedPlayersMyCombat() wasn't being called at end of combat. This made it possible to use warrior's stand in your own declare attackers step.
In fixing the bug I had to figure out where to put the call to clearAttackedPlayersMyCombat, and noticed that tracking for "this combat" effects was being cleared in multiple places, so I moved them together. 
Good call on the issue with things being removed from combat, I will fix that.